### PR TITLE
Local mocking from CLI

### PR DESCRIPTION
**Pull request by kylehodgetts**
This PR allows users to locally mock their apis using a mock server that returns responses based on content type examples or schemas.

If examples are specified, they are used and returned. If schemas are defined, the server will generate a response that fits the schema.

closes [#491](https://github.com/kubeshop/kusk-gateway/issues/491)

Screenshot 2022-07-18 at 14 48 41

## Checklist

- [x] tested locally
- [x] added new dependencies
- [x] updated the docs
- [ ] added a test

### CODE REVIEW

File: internal/mocking/server/server.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+type AccessLogEntry struct {
+	TimeStamp  string
+	Method     string
+	Path       string
+	StatusCode string
+}
+
+func New(ctx context.Context, client *client.Client, configFile, apiToMock string) (MockServer, error) {
+	const openApiMockImage = "muonsoft/openapi-mock:v0.3.1"
```
[the remark this answers is not in the document]
Comment (kylehodgetts) [reply internal/mocking/server/server.go:34]: This stricly uses docker so this is unnecessary. Furthermore the docker client cant find the image when it's docker.io/muonsoft/openapi-mock:v0.3.1
the following error is returned `unable to create mocking server: Error response from daemon: No such image: muonsoft/openapi-mock:v0.3.1`

**Review by jasmingacic — APPROVED** [review]
It looks good to me.

**Review by jasmingacic — APPROVED** [review]
Just a small comment otherwise it is looking good

File: cmd/mock.go [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+<doc>
+ <completed>true</completed>
+ <order>13</order>
+ <title>Mocked XML title</title>
+ <url>http://mockedURL.com</url>
+</doc>
+`,
+	Example: "kusk mock -i path-to-openapi-file.yaml",
+	Run: func(cmd *cobra.Command, args []string) {
+		homeDir, err := os.UserHomeDir()
```
Comment (povilasv) [inline cmd/mock.go:109]: small nit: Instead of writing config to homeDir maybe we can use tmp?

I imagine users dont need to care about this config?

**Review by povilasv — APPROVED** [review]

File: cmd/mock.go (hunk 2 above)
Comment (jasmingacic) [reply cmd/mock.go:109]: home dir isn't too bad if they want to reuse it and/or set some global variables there
I wouldn't touch it any further

File: internal/mocking/server/server.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+func (m MockServer) Start(ctx context.Context) (string, error) {
+	containerApiSpecPath := "mocking/fake-api.yaml"
+	containerMockingConfigFilePath := "/app/mocking/openapi-mock.yaml"
+
+	resp, err := m.client.ContainerCreate(
+		ctx,
+		&container.Config{
+			Image:        m.image,
+			ExposedPorts: nat.PortSet{"8080": struct{}{}},
```
Comment (mbana) [inline internal/mocking/server/server.go:66]: Can you not chose a randomly unused port like this <https://stackoverflow.com/a/43425461/16361002>?

Also, the repetition of `8080` is ideal. I'd use a variable.

**Review by mbana — APPROVED** [review]
LGTM.

**Pull request merged**